internal/hcp/api/client.go

```diff
@@ -44,39 +46,64 @@ func NewClient() (*Client, error) {
 		}
 	}
 
-	cl, err := httpclient.New(httpclient.Config{
+	hcpClientCfg := httpclient.Config{
 		SourceChannel: fmt.Sprintf("packer/%s", version.PackerVersion.FormattedVersion()),
-	})
-	if err != nil {
+	}
+	if err := hcpClientCfg.Canonicalize(); err != nil {
 		return nil, &ClientError{
 			StatusCode: InvalidClientConfig,
 			Err:        err,
 		}
 	}
 
+	cl, err := httpclient.New(hcpClientCfg)
+	if err != nil {
+		return nil, &ClientError{
+			StatusCode: InvalidClientConfig,
+			Err:        err,
+		}
+	}
 	client := &Client{
 		Packer:       packerSvc.New(cl, nil),
 		Organization: organizationSvc.New(cl, nil),
 		Project:      projectSvc.New(cl, nil),
 	}
+	//if both HCP_ORGANIZATION_ID and HCP_PROJECT_ID are set via env variables the hcpConfig may have all we need already.
+	if hcpClientCfg.Profile().OrganizationID != "" && hcpClientCfg.Profile().ProjectID != "" {
+		client.OrganizationID = hcpClientCfg.Profile().OrganizationID
+		client.ProjectID = hcpClientCfg.Profile().ProjectID
 
-	if err := client.loadOrganizationID(); err != nil {
-		return nil, &ClientError{
-			StatusCode: InvalidClientConfig,
-			Err:        err,
+		return client, nil
+	}
+
+	if client.OrganizationID == "" {
+		err := client.loadOrganizationID()
+		if err != nil {
+			return nil, &ClientError{
+				StatusCode: InvalidClientConfig,
+				Err:        err,
+			}
 		}
 	}
-	if err := client.loadProjectID(); err != nil {
-		return nil, &ClientError{
-			StatusCode: InvalidClientConfig,
-			Err:        err,
+
+	if client.ProjectID == "" {
+		err := client.loadProjectID()
+		if err != nil {
+			return nil, &ClientError{
+				StatusCode: InvalidClientConfig,
+				Err:        err,
+			}
 		}
 	}
 
 	return client, nil
 }
 
 func (c *Client) loadOrganizationID() error {
+	if env.HasOrganizationID() {
```

Comment:
Regarding this check, doesn't that conflict with the comment left on line 71?

> if both HCP_ORGANIZATION_ID and HCP_PROJECT_ID are set via env variables the hcpConfig may have all we need already.

If the client's `Profile` has them set already via the environment variable, is there a reason why we check for this here?

Reply:
Looking at the initialization code done in `Canonicalize`, I don't see where the org/project ID are loaded from the environment, is it possible that the `Profile.OrganizationID` or `Profile.ProjectID` are never set beforehand? If this is so, the check on line 72 seems to be always false, is my assumption correct?

Reply:
Hey great question! This took me a little ride at first, as well, but if you take a deep dive into the Canonicalize call there logic that build the default UserProfile (`Profile()`) using the FromEnv configuration options. `FromEnv` will check the the local environment for a number of environment variables and set each one if the criteria for lading the env. variable is met. For some variables the rule is "If set use it" for others like HCP_ORGANIZATION_ID and HCP_PROJECT_ID they both must be set in order for the SDK to loading them from the env. variable. 

The SDK does not validate the values like we are doing for the Project ID but if it decides in the future to do some validation and our Packer user sets both values we would benefit from the SDK doing the lifting. This is why I added the logic  to check if set use. I'm deferring the loading to the SDK because it may help us in the future when it comes to handling these vars consistently between products. 

> If the client's Profile has them set already via the environment variable, is there a reason why we check for this here?

As for this question the check is a guard clause if they are both set then we return and we don't check if they are set again. If only one is set then we have to check which is set and validate both accordingly. 

Now Packer does not need the Organization ID set via a env. variable at this time. But I don't see any harm in adding it. In fact when I first wrote this code Packer loaded both but it was removed in favor of ListProjects and ListOrganization. Given the recent changes in service principles and how the platform is evolving my thinking is we should support the four main env variables until we can fully rely on the SDK for them. 

Please let me know if that clears things up for you. 

I will update the comment in the code to call out that they are only set by the SDK if both Project ID and Organization ID are set.